### PR TITLE
@W-15644208 Upgrade formula-engine to Java/Jakarta EE 9 namespace

### PR DESCRIPTION
**Pull request by bairenlong-sfdc**
Discussed with @jfreschl, we will merge this to main to make it easier to fix other bugs. 

**We will release from this branch (main-jakarta) to core/252, we will merge this to main after all is good with ee9**

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -22,8 +22,8 @@
       <artifactId>grammaticus</artifactId>
     </dependency>
     <dependency>
-      <groupId>javax.annotation</groupId>
```
Comment (bairenlong-sfdc) [inline api/pom.xml:25]: using jarkatar annotation package

File: api/src/main/java/com/force/formula/util/FormulaFieldReferenceImpl.java [hunk 2]
```diff
@@ -5,65 +5,64 @@
  */
 package com.force.formula.util;
 
-import javax.annotation.concurrent.Immutable;
-
 import com.force.formula.FormulaFieldReference;
+import com.google.errorprone.annotations.Immutable;
```
Comment (bairenlong-sfdc) [inline api/src/main/java/com/force/formula/util/FormulaFieldReferenceImpl.java:9]: use Immutable from google, the rest changes are from IDE

File: api/src/main/java/com/force/formula/util/FormulaTextUtil.java [hunk 3]
```diff
@@ -1,44 +1,49 @@
...
-
 import com.force.i18n.commons.text.DeferredStringBuilder;
 import com.force.i18n.commons.util.collection.IntHashMap;
 import com.google.common.base.CharMatcher;
 import com.google.common.base.Joiner;
 import com.google.common.escape.Escaper;
 import com.google.common.escape.Escapers;
+import com.google.errorprone.annotations.Immutable;
+
+import jakarta.annotation.Nullable;
```
Comment (bairenlong-sfdc) [inline api/src/main/java/com/force/formula/util/FormulaTextUtil.java:18]: using packages from jakarta & google, the rest are from IDE reformatting

File: pom.xml [hunk 4]
```diff
@@ -379,6 +382,29 @@
           </execution>
         </executions>
       </plugin>
+
+      <!-- temp to fix eclipse compiler error-->
```
Comment (bairenlong-sfdc) [inline pom.xml:386]: This is for eclipse on local only.

File: test-utils/pom.xml [hunk 5]
```diff
@@ -36,71 +36,65 @@
...
             <optional>true</optional>
-        </dependency>        
+        </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>gcloud</artifactId>
             <version>${testcontainers.version}</version>
             <optional>true</optional>
         </dependency>
-        <dependency>
```
Comment (bairenlong-sfdc) [inline test-utils/pom.xml:87]: moved to google-test to make test-utils EE9 ready

**Review by jfreschl — APPROVED** [review]

File: api/pom.xml [hunk 6]
```diff
@@ -9,7 +9,7 @@
   <parent>
    <groupId>com.salesforce.formula</groupId>
    <artifactId>formula-engine</artifactId>
-   <version>0.3.23-SNAPSHOT</version>
+   <version>0.9.0-SNAPSHOT</version>
```
Comment (bairenlong-sfdc) [inline api/pom.xml:12]: We will starting with version 3.9.0 for EE9

**Pull request merged**